cmake/kokkos_tribits.cmake

```diff
@@ -313,10 +318,18 @@ ENDFUNCTION()
 FUNCTION(KOKKOS_ADD_LIBRARY LIBRARY_NAME)
   IF (KOKKOS_HAS_TRILINOS)
     TRIBITS_ADD_LIBRARY(${LIBRARY_NAME} ${ARGN})
-     #Stolen from Tribits - it can add prefixes
+    #Stolen from Tribits - it can add prefixes
     SET(TRIBITS_LIBRARY_NAME_PREFIX "${${PROJECT_NAME}_LIBRARY_NAME_PREFIX}")
     SET(TRIBITS_LIBRARY_NAME ${TRIBITS_LIBRARY_NAME_PREFIX}${LIBRARY_NAME})
-    KOKKOS_SET_LIBRARY_PROPERTIES(${TRIBITS_LIBRARY_NAME} PLAIN_STYLE)
+    #Tribits has way too much techinical debt and baggage to even
+    #allow PUBLIC target_compile_options to be used. It forces C++ flags on projects
+    #as a giant blob of space-separated strings. We end up with duplicated
+    #flags between the flags implicitly forced on Kokkos-dependent and those Kokkos
+    #has in its public INTERFACE_COMPILE_OPTIONS.
+    #These do NOT get de-deduplicated because Tribits
+    #creates flags as a giant monolithic space-separated string
+    #Do not set any transitive properties and keep everything working as before
+    #KOKKOS_SET_LIBRARY_PROPERTIES(${TRIBITS_LIBRARY_NAME} PLAIN_STYLE PRIVATE)
```

Comment:
Do I understand correctly that this line is indeed not used and all the modifications to `KOKKOS_SET_LIBRARY_PROPERTIES` are not necessary at all?

Reply:
Yeah, I initially wanted the ability to add flags `PRIVATE`ly, but I can revert that.

Reply:
OK. If we don't use it, let's just revert it.